src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Source.cs

```diff
@@ -122,87 +123,102 @@ private static async Task<Checksum> ComputeSourceSymbolsChecksumAsync(ProjectSta
-                // Add all child members
+                // Walk the symbols with this name, and add all their child namespaces and types, grouping them together
+                // based on their name.  There may be multiple (for example, Action<T1>, Action<T1, T2>, etc.)
                 foreach (var symbol in symbolsWithSameName)
+                    AddChildNamespacesAndTypes(symbol, childSymbolsByName);
+
+                // Now, go through all those groups and make the single mapping from their name to the builder-node we
+                // just created above, and recurse into their children as well.
+                foreach (var (childName, childSymbols) in childSymbolsByName)
                 {
-                    lookup(symbol, symbolMap);
+                    seenNames.Add(childName);
+                    GenerateSourceNodes(childName, nodeIndex, childSymbols, list);
                 }
 
-                foreach (var (symbolName, symbols) in symbolMap)
-                    GenerateSourceNodes(symbolName, nodeIndex, symbols, list, lookup);
+                // The above loops only create nodes for namespaces and types.  we also want nodes for members as well.
+                // However, we do not want to force the symbols for those members to be created just to get the names.
+                //
+                // So walk through the symbols again, and for the named-types grab all the member-names contained
+                // therein.  If we didn't already see that child name when recursing above, then make a builder-node for
+                // it that points to the builder-node we just created above.
+
+                foreach (var symbol in symbolsWithSameName)
+                {
+                    if (symbol is INamedTypeSymbol namedType)
+                    {
+                        foreach (var childMemberName in namedType.MemberNames)
+                        {
+                            if (seenNames.Add(childMemberName))
+                                list.Add(new BuilderNode(childMemberName, nodeIndex));
+                        }
+                    }
+                }
             }
             finally
             {
-                FreeSymbolMap(symbolMap);
+                FreeSymbolMap(childSymbolsByName);
             }
         }
 
-        private static readonly Action<ISymbol, MultiDictionary<string, ISymbol>> s_getMembersNoPrivate =
-            (symbol, symbolMap) => AddSymbol(symbol, symbolMap, s_useSymbolNoPrivate);
-
-        private static void AddSymbol(ISymbol symbol, MultiDictionary<string, ISymbol> symbolMap, Func<ISymbol, bool> useSymbol)
+        private static void AddChildNamespacesAndTypes(INamespaceOrTypeSymbol symbol, MultiDictionary<string, INamespaceOrTypeSymbol> symbolMap)
         {
-            if (symbol is INamespaceOrTypeSymbol nt)
+            if (symbol is INamespaceSymbol namespaceSymbol)
             {
-                foreach (var member in nt.GetMembers())
```

Comment:
this was hte problem before.  GetMembers() would realize all symbols both for namespaces and for types.  now we only realize all children for namespaces (which is just other namespaces and types), but for types we only realize child types.

Reply:
makes sense